operations/helm/tests/test-oss-values-generated/mimir-distributed/templates/ingester/ingester-statefulset.yaml

```diff
@@ -45,6 +45,16 @@ spec:
         {}
       affinity:
         {}
+      topologySpreadConstraints:
+        
+        - maxSkew: 1
```

Comment:
I don't think this is correct for ingesters. We don't want to schedule 2 ingesters on the same node. Am I missing anything?

Reply:
previously we didn't have any affinity for the ingesters. I think the reason was so that the chart can run on a small (or laptop-local) cluster.

I added the TSC with a max skew of 1, which I think is a better state than before. Kubernetes would only start stacking ingesters if all available nodes already have one ingester. And you can still run the chart on a small cluster.

Reply:
I see you're point, but doesn't make jsonnet and helm diverging? 🤔

Reply:
the small.yaml, large.yaml, capped-small.yaml, and capped-large.yaml do have podAntiAffinity for the ingesters (on a different note, they are using the wrong label matchers, and I've opened #2681 about that)

Reply:
(posted last comment without seeing yours)

> I see your point, but doesn't make jsonnet and helm diverging? 🤔

It does... We don't have any concept of "profiles" or "environments" in jsonnet, do we? So we can't have the equivalent of small.yaml et al. in jsonnet.

As a contributor to the chart I can accept having to apply my own values yaml when working locally so the chart runs on a local cluster.

@krajorama @andyasp @johannaratliff do you folks have opinions on this?

Reply:
also including @lamida @zenador @Logiraptor wih the same question

Reply:
_In any case, don't block on my comment, but curious how this discussion unfolds._

Reply:
> As a contributor to the chart I can accept having to apply my own values yaml  

I agree that _as a contributor_ it's not a huge deal to need some specific values to make it run. I have a `dev.yaml` that sets everything up on my local cluster. I think this is a worse experience for new users, since they won't necessarily know how to make it work locally and they may give up before finding the information needed or asking questions in the community channels.

I think we have to choose what use-case we want the default values to target: laptop usage or production usage. So far we've targeted laptop usage.

> I see your point, but doesn't make jsonnet and helm diverging? 🤔

Yes, but at least for the CI step we can handle this by modifying the values.yaml used to generate the parity configs.

Reply:
Agree with @Logiraptor , I wouldn't add anti affinity rules to the default values. Before we had replication factor = 3 set by default, the chart started just one ingester so it worked out of the box for testing. I've removed the anti affinity rules when we went for 3x replication factor by default.

I think we should instead write a getting started guide for production use case.

Reply:
I wonder if we should ever do a switch to start supporting a prod-like deployment more than a local toy deployment out of the box. And if yes, then when? naively - when our users try mimir out first on a laptop, then install it in a cluster. We can do a survey in the community slack to get a show of hands?